server.py

```diff
@@ -5,11 +5,575 @@
+      },
+      {
+        "id": 27,
+        "restaurant_id": 9,
+        "name": "Sally",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 4,
+        "comments": "One of the tastiest and most unique ramen places I've been to in NYC, but also the priciest. I think overall its worth the try. Not an everyday casual ramen shop though."
+      },
+      {
+        "id": 28,
+        "restaurant_id": 10,
+        "name": "Steve",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 5,
+        "comments": "Head to this laid-back Long Island City restaurant for beef tongue tacos, chicken smothered in a heady mole sauce, and a monster crab tostada. New York's only Michelin-starred Mexican restaurant is an especially cool choice for lunch during the week or drinks after work. Eater critic Ryan Sutton awarded this restaurant two stars."
+      },
+      {
+        "id": 29,
+        "restaurant_id": 10,
+        "name": "Rob",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 5,
+        "comments": "The hype was real. Please go. Get the ceviche. And the tres leches. You're welcome"
+      },
+      {
+        "id": 30,
+        "restaurant_id": 10,
+        "name": "Jason",
+        "createdAt": 1504095567183,
+        "updatedAt": 1504095567183,
+        "rating": 4,
+        "comments": "For a Michelin star restaurant, it's fairly priced and the food is fairly good. Started with a strawberry margarita which was good in flavor but not much alcohol. Had the chicken enchiladas with salsa verde and it was really good. Great balance in flavor and a good portion. Extra tasty with their hot sauces. My wife had the lamb but it was a bit too salty for our taste. Although, it was cooked very well and fell off the bone. The highlight of the night was the tres leches cake. Probably the best I've ever had to be honest. Not too sweet and very milky. Overall, one of my top 3 favorite Mexican in NY."
+      }]
+
+
+@app.route('/restaurants', methods=['GET'])
+def getAllRestaurants():
+    return jsonify(restaurants)
+
+@app.route('/reviews', methods=['GET'])
+def getAllReviews():
+    return jsonify(reviews)
+
+@app.route('/restaurants/favorites', methods=['GET'])   
+def getFavoritedRestaurants():
+  results = []
+  for restaurant in restaurants:
+    if(restaurant.get("is_favorite") == True):
+      results.append(restaurant)
+  return jsonify(results)    
+
+@app.route('/restaurants/<int:restaurant_id>', methods=['GET']) 
+def getRestaurantByID(restaurant_id):
+  for restaurant in restaurants:
+    if(restaurant.get("id") == restaurant_id):
+      return jsonify(restaurant)  
+
+@app.route('/reviews/restaurant/<int:restaurant_id>', methods=['GET'])
+def getAllReviewsForRestaurant(restaurant_id):
+  results = []
+  for review in reviews:
```

Comment:
This section of code is fine, but it could be simpler using list comprehensions. It is the same code either way, but you might find this useful, so another way to represent this code is. List comprehensions can do more than just this, so it is worth checking them out (or at least being aware of them)
`return jsonify( [review for review in reviews if review.get("restaurant_id") == restaurant_id] )`
This style of coding comes from functional programming, and seems to be propagating into every other language

Reply:
I will look into this. Seems interesting